doc/src/manual/methods.md

```diff
@@ -515,6 +515,225 @@ julia> wait(schedule(t, 1))
 "definition for Int"
 ```
 
+## Design Patterns with Parametric Methods
+
+
+While complex dispatch logic is not required for performance or usability,
+sometimes it can be the best way to express some algorithm.
+Here are a few common design patterns that come up sometimes when using dispatch in this way.
+
+### Extracting the type parameter from a super-type
+
+
+Here is the correct code template for returning the element-type `T`
+of any arbitrary subtype of `AbstractArray`:
+
+```julia
+abstract AbstractArray{T, N}
+eltype(::Type{A}) where A<:AbstractArray{T} where T = T
```

Comment:
@fredrikekre I find that syntax a bit confusing.  In fact it seems a bit reminiscent of the oldern days (i.e. recently) when we had `eltype{T,A<:AbstractArray{T}}(::Type{A}) = T`.  So somehow it doesn't mix so well with the `where`.  Do you know what the advantage is to have both types of syntax?

This works too: `eltype(::Type{<:AbstractArray{T}}) where {T} = T`, maybe nicest?